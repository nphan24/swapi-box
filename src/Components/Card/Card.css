@import url('https://fonts.googleapis.com/css?family=IBM+Plex+Sans:400,600');

.people-card {
  width: 260px;
  min-width: 220px;
  border: 2px solid black;
  min-height: 350px;
  margin-top: 15px;
  border-radius: 30px 30px 30px 30px;
  background-image: url('../../images/background.jpg');
  background-size: cover;
  color: white;
  font-family: 'IBM Plex Sans', sans-serif;
  font-weight: 400;
  font-size: 1.2em;
}

.card-favorite-button {
  width: 80px;
  height: 30px;
  font-size: .8em;
  border-radius: 30px 30px 30px 30px;
  margin-top: 15px;
}

.card-favorite-button:hover {
  background-color: #E5EDFC;
}

.card-info {
  margin: 0;
}