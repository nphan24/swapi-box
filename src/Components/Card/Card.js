import React from 'react';
import './Card.css';

const Card = ({name, homeworld, population, species}) => {
  return (
    <div>
      <h3>{name}</h3>
      <h4>{homeworld}</h4>
      <p>{population}</p>
      <p>{species}</p>
    </div>
  );
};

export default Card;