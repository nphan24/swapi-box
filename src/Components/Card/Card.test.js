import React from 'react';
import Card from './Card';
import { shallow } from 'enzyme';

describe('Card', () => {
  let wrapper;
  let mockObject;
  let mockFunction;

  beforeEach(()=> {
    mockObject = {
      name: 'Cat',
      homeworld: 'Tatooine',
      population: '5'
    };
    mockFunction = jest.fn();
    wrapper = shallow(<Card info={mockObject} setFavorites={mockFunction} />);
  });

  it('should exist', () => {
    expect(wrapper).toBeDefined();
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should create a new card for every element with the correct number of keys', () => {
    expect(wrapper.find('p').length).toEqual(2);
  });

  it('should call setFavorites on click', () => {
    wrapper.find('button').simulate('click');
    expect(setFavorites).toHaveBeenCalledWith(mockObject);
  });
});