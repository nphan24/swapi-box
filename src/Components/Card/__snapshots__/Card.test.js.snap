// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card
    info={
        Object {
            "homeworld": "Tatooine",
            "name": "Cat",
            "population": "5",
          }
    }
    setFavorites={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h3
          className="object-name"
>
          Cat
</h3>,
        Array [
          <p
            className="card-info"
>
            homeworld
            : 
            Tatooine
</p>,
          <p
            className="card-info"
>
            population
            : 
            5
</p>,
        ],
        <button
          className="card-favorite-button undefined"
          onClick={[Function]}
>
          Favorites
</button>,
      ],
      "className": "each-card",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Cat",
          "className": "object-name",
        },
        "ref": null,
        "rendered": "Cat",
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "homeworld",
            ": ",
            "Tatooine",
          ],
          "className": "card-info",
        },
        "ref": null,
        "rendered": Array [
          "homeworld",
          ": ",
          "Tatooine",
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "population",
            ": ",
            "5",
          ],
          "className": "card-info",
        },
        "ref": null,
        "rendered": Array [
          "population",
          ": ",
          "5",
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Favorites",
          "className": "card-favorite-button undefined",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Favorites",
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h3
            className="object-name"
>
            Cat
</h3>,
          Array [
            <p
              className="card-info"
>
              homeworld
              : 
              Tatooine
</p>,
            <p
              className="card-info"
>
              population
              : 
              5
</p>,
          ],
          <button
            className="card-favorite-button undefined"
            onClick={[Function]}
>
            Favorites
</button>,
        ],
        "className": "each-card",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Cat",
            "className": "object-name",
          },
          "ref": null,
          "rendered": "Cat",
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "homeworld",
              ": ",
              "Tatooine",
            ],
            "className": "card-info",
          },
          "ref": null,
          "rendered": Array [
            "homeworld",
            ": ",
            "Tatooine",
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "population",
              ": ",
              "5",
            ],
            "className": "card-info",
          },
          "ref": null,
          "rendered": Array [
            "population",
            ": ",
            "5",
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Favorites",
            "className": "card-favorite-button undefined",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Favorites",
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
